File: rabbitmq-poc/src/main/java/com/example/demo/MessageListener.java
package com.example.demo;

import org.springframework.amqp.rabbit.annotation.RabbitListener;
import org.springframework.stereotype.Component;

@Component
public class MessageListener {

    private final MessageRepository repository;

    public MessageListener(MessageRepository repository) {
        this.repository = repository;
    }

    @RabbitListener(queues = "${app.queue.name}")
    public void receive(String content) {
        repository.save(new MessageEntity(content));
    }
}

File: rabbitmq-poc/src/main/java/com/example/demo/MessageListener.kt
package com.example.demo

import org.springframework.amqp.rabbit.annotation.RabbitListener
import org.springframework.stereotype.Component

@Component
class MessageListener(private val repository: MessageRepository) {
    @RabbitListener(queues = ["\${app.queue.name}"])
    fun receive(content: String) {
        repository.save(MessageEntity(content = content))
    }
}
